crates/trie/sparse-parallel/src/trie.rs

```diff
@@ -32,6 +32,47 @@ pub const NUM_LOWER_SUBTRIES: usize = 16usize.pow(UPPER_TRIE_MAX_DEPTH as u32);
 
 /// A revealed sparse trie with subtries that can be updated in parallel.
 ///
+/// ## Structure
+///
+/// The trie is divided into two tiers for efficient parallel processing:
+/// - **Upper subtrie**: Contains nodes with paths shorter than `UPPER_TRIE_MAX_DEPTH` (2 nibbles)
+/// - **Lower subtries**: An array of 256 subtries, each handling nodes with paths of 2+ nibbles
+///
+/// Node placement is determined by path depth:
+/// - Paths with < 2 nibbles go to the upper subtrie
```

Comment:
I wonder if we should use the `UPPER_TRIE_MAX_DEPTH` const here just so that we're consistent about either using it, or just `2`